proposals/semi-auto-properties.md

```diff
@@ -1,14 +1,325 @@
-# Semi-auto-properties (a.k.a. `field` keyword in properties)
+# `field` keyword in properties
 
 ## Summary
-Extend auto-properties to allow them to still have an automatically generated backing field, while still allowing for bodies to be provided for accessors.  Auto-properties can also use a new contextual `field` keyword in their body to refer to the auto-prop field.
+
+Extend all properties to allow them to reference an automatically generated backing field using the new contextual keyword `field`. Properties may now also contain an accessor _without_ a body alongside an accessor _with_ a body.
 
 ## Motivation
-Standard auto-properties only allow for setting or getting the backing field directly, giving some control only by access modifying the accessor methods. Sometimes there is more need to have control over what happens when accessing an auto-property, without being confronted with all overhead of a standard property.
 
-Two common scenarios are that you want to apply a constraint on the setter, ensuring the validity of a value. The other being raising an event that informs about the property going to be changed/having been changed.
+Auto properties only allow for directly setting or getting the backing field, giving some control only by placing access modifiers on the accessors. Sometimes there is a need to have additional control over what happens in one or both accessors, but this confronts users with the overhead of declaring a backing field. The backing field name must then be kept in sync with the property, and the backing field is scoped to the entire class which can result in accidental bypassing of the accessors from within the class.
+
+There are two common scenarios in particular: applying a constraint on the setter to ensure the validity of a value, and raising an event such as `INotifyPropertyChanged.PropertyChanged`.
```

Comment:
nit: add a common scenario of "returning a default value in the case where a property has never been given a value".

Reply:
primarily because this seems to only indicate value for `field` in `set;` and i'd like us to indicate value in `get;` as well.